src/wallet/test/fuzz/scriptpubkeyman.cpp

```diff
@@ -200,5 +207,129 @@ FUZZ_TARGET(scriptpubkeyman, .init = initialize_spkm)
+                    [&] {
+                        script = GetScriptForDestination(WitnessV0KeyHash(pub_key));
+                    },
+                    [&] {
+                        script = ConsumeScript(fuzzed_data_provider);
+                    }
+                );
+                (void)legacy_data.AddCScript(script);
+            },
+            [&] {
+                CKey key;
+                if (!keys.empty()) {
+                    key = PickValue(fuzzed_data_provider, keys);
+                } else {
+                    key = ConsumePrivateKey(fuzzed_data_provider);
+                }
+                if (!key.IsValid()) return;
+                const auto num_keys{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(1, MAX_PUBKEYS_PER_MULTISIG)};
+                std::vector<CPubKey> pubkeys;
+                for (size_t i = 0; i < num_keys; i++) {
+                    if (fuzzed_data_provider.ConsumeBool()) {
+                        pubkeys.emplace_back(key.GetPubKey());
+                    } else {
+                        CKey private_key{ConsumePrivateKey(fuzzed_data_provider)};
+                        if (!private_key.IsValid()) return;
+                        pubkeys.emplace_back(private_key.GetPubKey());
+                    }
+                }
+                if (pubkeys.size() < num_keys) return;
+                const auto multisig_script{GetScriptForMultisig(num_keys, pubkeys)};
+                (void)legacy_data.AddCScript(multisig_script);
+            }
+        );
+    }
+
+    std::optional<MigrationData> res{legacy_data.MigrateToDescriptor()};
+    if (add_hd_chain) {
+        CExtKey master_key;
+        master_key.SetSeed(hd_key);
+        assert(res->master_key == master_key);
+    }
+
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool() && good_data, 50) {
+        std::optional<CMutableTransaction> mtx{ConsumeDeserializable<CMutableTransaction>(fuzzed_data_provider, TX_WITH_WITNESS)};
+        if (!mtx) {
+            good_data = false;
+            return;
+        }
+        if (!mtx->vout.empty() && fuzzed_data_provider.ConsumeBool()) {
+            const auto idx{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, mtx->vout.size() - 1)};
+            const auto output_type{fuzzed_data_provider.PickValueInArray({OutputType::BECH32, OutputType::LEGACY})};
+            const auto label{fuzzed_data_provider.ConsumeRandomLengthString()};
+            CScript spk{GetScriptForDestination(*Assert(wallet.GetNewDestination(output_type, label)))};
+            mtx->vout[idx].scriptPubKey = spk;
+        }
+        wallet.AddToWallet(MakeTransactionRef(*mtx), TxStateInactive{}, /*update_wtx=*/nullptr, /*fFlushOnClose=*/false, /*rescanning_old_block=*/true);
+    }
+
+    WalletBatch batch{wallet.GetDatabase()};
+    LOCK(wallet.cs_wallet);
+    CBlockLocator loc{chainstate.m_chain.GetLocator()};
+    assert(batch.WriteBestBlock(loc));
+    (void)wallet.ApplyMigrationData(batch, *res);
```

Comment:
You cannot call `ApplyMigrationData` without executing the entire `DoMigration step`. Tat's just incorrect. This code skips the creation of the watch-only and solvable wallets from the `MigrationData` result, causing a crash inside `ApplyMigrationData`.

Reply:
Yes, just realized it. I'm going to change it, but afaik `DoMigration` can't be called directly here.

Reply:
Done